File: go/problem/car_fleet.go
package problem

import "sort"

func carFleet(target int, position []int, speed []int) int {
	var cars = make([][2]float64, len(position))
	var result int
	var time float64
	for i := range cars {
		cars[i] = [2]float64{
			float64(position[i]),
			// time
			float64((target - position[i]) / speed[i]),
		}
	}
	sort.Slice(cars, func(i, j int) bool {
		return cars[i][0] > cars[j][0]
	})
	for _, t := range cars {
		if t[1] > time {
			time = t[1]
			result++
		}
	}
	return result
}

File: rust/src/problems/car_fleet.rs
pub struct Solution;

impl Solution {
    pub fn car_fleet(target: i32, position: Vec<i32>, speed: Vec<i32>) -> i32 {
        let mut cars: Vec<(i32, f64)> = position
            .iter()
            .zip(speed.iter())
            .map(|(&p, &s)| (p, (target - p) as f64 / s as f64))
            .collect();
        cars.sort_unstable_by(|a, b| b.0.cmp(&a.0));
        let mut result = 0;
        let mut time = 0.0;
        for (_, t) in cars {
            if t > time {
                time = t;
                result += 1;
            }
        }
        result
    }
}
